subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -206,16 +208,47 @@ uint8_t ll_sync_create_cancel(void **rx)
 	return 0;
 }
 
-uint8_t ll_sync_terminate(uint16_t handle)
+uint8_t ll_sync_terminate(uint16_t handle, void **rx)
 {
+	struct node_rx_pdu *node_rx;
+	memq_link_t *link_sync_lost;
 	struct ll_sync_set *sync;
+	uint32_t volatile ret_cb;
```

Comment:
Why does variable need to be volatile? 

It seems that it is being cast to a `void *` (i.e. a non-volatile type) further down.

Reply:
`ull_ticker_status_give` updates ticker operation status value in `ret_cb` in ISR, and `ull_ticker_status_take` checks this status in thread context.

It is type cast to (void *) when passing to ticker API as its just used as parameter when calling the ticker operations callback. The callback in this case `ull_ticker_status_give` uses it as `uint32_t volatile *`. https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/ull.c#L1070